src/lib/twitter/notifyTwitter.ts

```diff
@@ -0,0 +1,23 @@
+import TwitterAPI from 'twitter-api-v2';
+import { NFTSale } from "lib/marketplaces";
+import axios from 'axios';
+
+export default async function notifyTwitter(twitterClient: TwitterAPI, nftSale: NFTSale) {
+    const hasImage = Boolean(nftSale.nftData?.image);
+    const data = hasImage ? await getImageDataFromUrl(nftSale.nftData?.image as string) : null;
+    const nftName = nftSale.nftData?.name;
+    const text = `${nftName} has just been sold for ${nftSale.getPriceInSOL()} S◎L at ${nftSale.marketplace.name}! #SolanaNFTs #NFTSale`
+    const mediaArr: string[] = [];
+    if (hasImage) {
+        const media = await twitterClient.v1.uploadMedia(data as Buffer, { type: 'gif' });
```

Comment:
I am concern about the performance of uploading the image for each notification. It might take a long time and may not scale if there's an influx of notifications. 
Is it possible to just post the link to the image and let twitter do the preview?